### PR TITLE
isEpochTimestamp fix

### PR DESCRIPTION
**Pull request by Manuel-777**
I am not sure how non patreons could have ended with epoch timestamps unless we were forcefully downgrading Js times into regular Epoch (I can see myself doing that in the past, so why not)

Anyhow, this should fix the display issues in #1022 and include a new handy method that should probably be used just in case for a while in the rest of the datasets.

### CODE REVIEW

File: src/shared/util.ts [hunk 1]
```diff
@@ -559,3 +559,14 @@ export function prettierDeckData(deckData: InternalDeck): InternalDeck {
...
+// Im not sure if this would be better as a converter instead of a check
+export function isEpochTimestamp(timestamp: number): boolean {
+  const asDate = new Date(timestamp);
+  // Even if we have a date in epoch between 2000 and
+  // 2100 it will be parsed as 1970.
+  if (asDate && asDate.getUTCFullYear() < 2000) {
+    return true;
+  }
+  return false;
+}
```
Comment (Thaoden) [inline src/shared/util.ts:572]:
```suggestion
return asDate.getUTCFullYear() < 2000;
```

File: src/window_background/httpApi.ts [hunk 2]
```diff
@@ -84,6 +84,7 @@ function syncUserData(data: any): void {
       const id = doc._id;
       doc.id = id;
       delete doc._id;
+      if (isEpochTimestamp(doc.date)) doc.date = doc.date * 1000;
```
Comment (Thaoden) [inline src/window_background/httpApi.ts:87]:
```suggestion
if (isEpochTimestamp(doc.date)) doc.date *= 1000;
```

File: src/window_background/loadPlayerConfig.ts [hunk 3]
```diff
@@ -134,6 +135,9 @@ export async function loadPlayerConfig(): Promise<void> {
     const eventsList: InternalEvent[] = savedData.courses_index
       .filter((id: string) => savedData[id])
       .map((id: string) => {
+        if (isEpochTimestamp(savedData[id].date)) {
+          savedData[id].date = savedData[id].date * 1000;
+        }
```
Comment (Thaoden) [inline src/window_background/loadPlayerConfig.ts:140]:
```suggestion
        if (isEpochTimestamp(savedData[id].date)) {
          savedData[id].date *= 1000;
        }
```

I think the `*=` operator exists in TS, too?

**Comment by Thaoden** [comment]
Looks good, though the issue did not happen to all non-Patreons.

Also, are the new timestamps persisted in the local user database?

**Comment by Manuel-777** [comment]
> Also, are the new timestamps persisted in the local user database?

Not unless they synchronize or the event has not finished yet

File: src/shared/util.ts [hunk 4]
```diff
@@ -565,8 +565,5 @@ export function isEpochTimestamp(timestamp: number): boolean {
   const asDate = new Date(timestamp);
   // Even if we have a date in epoch between 2000 and
   // 2100 it will be parsed as 1970.
-  if (asDate && asDate.getUTCFullYear() < 2000) {
-    return true;
-  }
-  return false;
+  return asDate && asDate == asDate && asDate.getUTCFullYear() < 2000;
```
Comment (Thaoden) [inline src/shared/util.ts:568]: Why the equality check?

**Pull request merged**